ARCHITECTURE.md

```diff
@@ -1,113 +1,265 @@
 
 ## Transports
-Currently two network transport mechanisms are supported: 
+
+Currently two network transport mechanisms are supported:
+
 - IP based
+
 - Unity Relay Based
 
 In the first, the clients connect directly to a host via IP address. This will only work if both are in the same local area network or if the host forwards ports.
 
-For Unity Relay based multiplayer sessions, some setup is required. Please see our guide [here](Documentation/Unity-Relay/README.md). 
+For Unity Relay based multiplayer sessions, some setup is required. Please see our guide [here](Documentation/Unity-Relay/README.md).
 
 Please see [Multiplayer over internet](README.md) section of our Readme for more information on using either one.
 
-The transport is set in the transport field in the `NetworkManager`. We are using the following transport:
-- **Unity Transport Package (UTP):** Unity Transport Package is a network transport layer, packaged with network simulation tools which are useful for spotting networking issues early during development. This protocol is initialized to use direct IP to connect, but is configured at runtime to use Unity Relay if starting a game as a host using the Lobby Service, or joining a Lobby as a client. Unity Relay is a relay service provided by Unity services, supported by Unity Transport. See the documentation on [Unity Transport Package](https://docs-multiplayer.unity3d.com/docs/transport-utp/about-transport-utp/#unity-transport-package-utp) and on [Unity Relay](https://docs-multiplayer.unity3d.com/docs/relay/relay).
+The transport is set in the transport field in the `NetworkManager`. We are using __Unity Transport Package (UTP)__.
 
-To add new transports in the project, parts of `GameNetPortal` and `ClientGameNetPortal` (transport switches) need to be extended.
+Unity Transport Package is a network transport layer, packaged with network simulation tools which are useful for spotting networking issues early during development. This protocol is initialized to use direct IP to connect, but is configured at runtime to use Unity Relay if starting a game as a host using the Lobby Service, or joining a Lobby as a client. Unity Relay is a relay service provided by Unity services, supported by Unity Transport. See the documentation on [Unity Transport Package](https://docs-multiplayer.unity3d.com/docs/transport-utp/about-transport-utp/#unity-transport-package-utp) and on [Unity Relay](https://docs-multiplayer.unity3d.com/docs/relay/relay).
 
-## Game state / Scene flow
-In Boss Room, scenes correspond to top-level Game States (see `GameStateBehaviour` class) in a 1:1 way. That is, there is a `MainMenu` scene, `Character Select` scene (and state), and so on. 
+`StartingHostState` and `ClientConnectingState` are the ones that would need to be extended if we needed to add a new transport. Both of these classes assume that we are using UTP.
```

Comment:
```suggestion
```
this is talking about states before the state machine is introduced. moving to a bit lower in the file.